econia-labs/emojicoin-dot-fun
# [ECO-2620] View functions, preliminary testing

# Background

I initially began working on this PR while its base branch from #478 was
awaiting review. In doing so I began to make some changes to production logic
that I noticed as a result of starting to write tests.

However, in parallel #478 underwent considerable review and so after finalizing
changes there, I rebased this PR so that it could merge in the logic changes
that I had started working on, which were created as a result of test fixturing.
The result is a final prototype complete with view functions, that can then be
merged in a final review of #478.

# Changes

1. Add view functions and return unpackers for all core structs.
1. Add test-only fixturing to enable comprehensive state/event mocking.

# Test code

1. Add assorted mocking assertion functions.
1. Add additional test coin factories for `init_module` happy path tests.
1. Add `init_module` happy path tests for assorted pseudo-random seeds.
1. Add admin function happy path and expected failure tests

# Testing

From in `src/move/emojicoin_arena`:

```sh
git ls-files | entr -c sh -c " \
       aptos move test --coverage --dev --move-2  &&
       aptos move fmt \
"
```
